File: testdata/jennies/builders/known_any/TypescriptBuilder/src/known_any/somestruct_builder_gen.ts
import * as cog from '../cog';
import * as known_any from '../known_any';

export class SomeStructBuilder implements cog.Builder<known_any.SomeStruct> {
    private readonly internal: known_any.SomeStruct;

    constructor() {
        this.internal = known_any.defaultSomeStruct();
    }

    build(): known_any.SomeStruct {
        return this.internal;
    }

    title(title: string): this {
        if (!this.internal.config) {
            this.internal.config = known_any.defaultConfig();
        }
        this.internal.config.title = title;
        return this;
    }
}

File: testdata/jennies/builders/known_any/GoBuilder/known_any/somestruct_builder_gen.go
package known_any

import (
	cog "github.com/grafana/cog/generated/cog"
)

var _ cog.Builder[SomeStruct] = (*SomeStructBuilder)(nil)

type SomeStructBuilder struct {
    internal *SomeStruct
    errors cog.BuildErrors
}

func NewSomeStructBuilder() *SomeStructBuilder {
	resource := NewSomeStruct()
	builder := &SomeStructBuilder{
		internal: resource,
		errors: make(cog.BuildErrors, 0),
	}

	return builder
}



func (builder *SomeStructBuilder) Build() (SomeStruct, error) {
	if err := builder.internal.Validate(); err != nil {
		return SomeStruct{}, err
	}
	
	if len(builder.errors) > 0 {
	    return SomeStruct{}, cog.MakeBuildErrors("known_any.someStruct", builder.errors)
	}

	return *builder.internal, nil
}

func (builder *SomeStructBuilder) Title(title string) *SomeStructBuilder {
if builder.internal.Config == nil {
    builder.internal.Config = NewConfig()
}
    builder.internal.Config.(*Config).Title = title

    return builder
}

